service/controller/v22/resource/s3object/current.go

```diff
@@ -26,6 +27,24 @@ func (r *Resource) GetCurrentState(ctx context.Context, obj interface{}) (interf
 		return nil, microerror.Mask(err)
 	}
 
+	// During deletion, it might happen that the encryption key got already
+	// deleted. In such a case we do not have to do anything here anymore. The
+	// desired state computation usually requires the encryption key to come up
+	// with the deletion state, but in case it is gone we do not have to do
+	// anything here anymore. The current implementation relies on the bucket
+	// deletion of the s3bucket resource, which deletes all S3 objects and the
+	// bucket itself.
```

Comment:
Can somebody confirm this is correct? Maybe @ferrandinand?

Reply:
I pinged the wrong guy. 🙈 /cc @pipo02mix

Reply:
I would say yes but I am not the coder behind it, neither an expert. I think @rossf7 has better understanding here